classes/privacy/provider.php

```diff
@@ -0,0 +1,736 @@
+                'message'    => 'privacy:metadata:moodleoverflow_posts:message'
+            ],
+            'privacy:metadata:moodleoverflow_posts');
+
+        $collection->add_database_table(
+            'moodleoverflow_read',
+            [
+                'userid'       => 'privacy:metadata:moodleoverflow_read:userid',
+                'discussionid' => 'privacy:metadata:moodleoverflow_read:discussionid',
+                'postid'       => 'privacy:metadata:moodleoverflow_read:postid',
+                'firstread'    => 'privacy:metadata:moodleoverflow_read:firstread',
+                'lastread'     => 'privacy:metadata:moodleoverflow_read:lastread'
+            ],
+            'privacy:metadata:moodleoverflow_read');
+
+        $collection->add_database_table(
+            'moodleoverflow_subscriptions',
+            [
+                'userid'         => 'privacy:metadata:moodleoverflow_subscriptions:userid',
+                'moodleoverflow' => 'privacy:metadata:moodleoverflow_subscriptions:moodleoverflow'
+            ],
+            'privacy:metadata:moodleoverflow_subscriptions');
+
+        $collection->add_database_table(
+            'moodleoverflow_discuss_subs',
+            [
+                'userid'     => 'privacy:metadata:moodleoverflow_discuss_subs:userid',
+                'discussion' => 'privacy:metadata:moodleoverflow_discuss_subs:discussion',
+                'preference' => 'privacy:metadata:moodleoverflow_discuss_subs:preference'
+            ],
+            'privacy:metadata:moodleoverflow_discuss_subs');
+
+        $collection->add_database_table(
+            'moodleoverflow_ratings',
+            [
+                'userid'      => 'privacy:metadata:moodleoverflow_ratings:userid',
+                'postid'      => 'privacy:metadata:moodleoverflow_ratings:postid',
+                'rating'      => 'privacy:metadata:moodleoverflow_ratings:rating',
+                'firstrated'  => 'privacy:metadata:moodleoverflow_ratings:firstrated',
+                'lastchanged' => 'privacy:metadata:moodleoverflow_ratings:lastchanged'
+            ],
+            'privacy:metadata:moodleoverflow_ratings');
+
+        $collection->add_database_table(
+            'moodleoverflow_tracking',
+            [
+                'userid'           => 'privacy:metadata:moodleoverflow_tracking:userid',
+                'moodleoverflowid' => 'privacy:metadata:moodleoverflow_tracking:moodleoverflowid'
+            ],
+            'privacy:metadata:moodleoverflow_tracking');
+
+        $collection->link_subsystem('core_files',
+            'privacy:metadata:core_files'
+        );
+
+        return $collection;
+    }
+
+    /**
+     * Get the list of contexts that contain user information for the specified user.
+     *
+     * @param int $userid The user to search.
+     *
+     * @return contextlist $contextlist The list of contexts used in this plugin.
+     */
+    public static function _get_contexts_for_userid(int $userid) {
```

Comment:
`int` argument type is not supported in PHP 5.6, similar to return types (also affects many other instances in this class!)